javascripts/discourse/api-initializers/discourse-mermaid-theme-component.js

```diff
@@ -58,9 +59,55 @@ async function applyMermaid(element, key = "composer") {
       mermaid.dataset.processed = true;
       mermaid.querySelector(".spinner")?.remove();
     }
+
+    if (key === "composer") {
+      debounce(() => updateMarkdownHeight(mermaid, index), 500);
+    }
   });
 }
 
+function updateMarkdownHeight(mermaid, index) {
+  let height = parseInt(mermaid.getBoundingClientRect().height);
+  let calculatedHeight = parseInt(mermaid.dataset.calculatedHeight);
+
+  if (height === 0) {
+    return;
+  }
+
+  if (height !== calculatedHeight) {
+    mermaid.dataset.calculatedHeight = height;
+    // TODO: need to use API here
+    let composer = document.getElementsByClassName("d-editor-input")[0];
+    let old = composer.value;
+
+    let split = old.split("\n");
+
+    let n = 0;
+    for (let i = 0; i < split.length; i++) {
+      if (split[i].match(/```mermaid/)) {
+        if (n === index) {
+          split[i] = "```mermaid height=" + height;
```

Comment:
Are there other options for mermaid that could be present here that we are wiping out? Or is `height` the only one?

Reply:
height is all we got as far as I know.